File: Python/Data Structures & Algorithms Module - Python/Week5/QueueApp.py
"""
QueueApp.py
Billy Cussen
13/11/2020
"""
from Queue import Queue

myQueueApp = Queue()

try:
    print("Empty: "+str(myQueueApp.isEmpty()))
except Exception as e:
    print(e)
finally:
    try:
        print("Size: "+str(myQueueApp.getSize()))
    except Exception as e:
        print(e)
    finally:
        try:
            print("Peek: "+str(myQueueApp.peek()))
        except Exception as e:
            print(e)
        finally:
            try:
                print("DeQueue: "+str(myQueueApp.deQueue()))
            except Exception as e:
                print(e)
            finally:
                try:
                    print(myQueueApp.toString())
                except Exception as e:
                    print(e)

myQueueApp.enQueue(2)
myQueueApp.enQueue(4)
myQueueApp.enQueue(6)
myQueueApp.enQueue(8)
myQueueApp.enQueue(10)
myQueueApp.enQueue(12)

try:
    print("Empty: "+str(myQueueApp.isEmpty()))
except Exception as e:
    print(e)
finally:
    try:
        print("Size: "+str(myQueueApp.getSize()))
    except Exception as e:
        print(e)
    finally:
        try:
            print("Peek: "+str(myQueueApp.peek()))
        except Exception as e:
            print(e)
        finally:
            try:
                print("DeQueue: "+str(myQueueApp.deQueue()))
            except Exception as e:
                print(e)
            finally:
                try:
                    print(myQueueApp.toString())
                except Exception as e:
                    print(e)
                        
while myQueueApp.isEmpty() == False:
    print("DeQueue: "+str(myQueueApp.deQueue()))

try:
    print("Empty: "+str(myQueueApp.isEmpty()))
except Exception as e:
    print(e)
finally:
    try:
        print("Size: "+str(myQueueApp.getSize()))
    except Exception as e:
        print(e)
    finally:
        try:
            print("Peek: "+str(myQueueApp.peek()))
        except Exception as e:
            print(e)
        finally:
            try:
                print("DeQueue: "+str(myQueueApp.deQueue()))
            except Exception as e:
                print(e)
            finally:
                try:
                    print(myQueueApp.toString())
                except Exception as e:
                    print(e)
                finally:
                    myQueueApp = None

File: Java/Data Structures & Algorithms Module - Java/Week8/Revision/Week5/QueueApp.java
/*
QueueApp.java
Billy Cussen
13/11/2020
*/

public class QueueApp{

    public static void main(String[] args){

        QueueInterface<Integer> myQueue = new Queue<Integer>();

        try{
            System.out.println("Empty: "+myQueue.isEmpty());
        }catch(Exception e){
            System.out.println(e.getMessage());
        } finally{
            try{
                System.out.println("Size: "+myQueue.size());
            }catch(Exception e){
                System.out.println(e.getMessage());
            } finally{
                try{
                    System.out.println("Dequeue: "+myQueue.deQueue());
                }catch(Exception e){
                    System.out.println(e.getMessage());
                } finally{
                    try{
                        System.out.println(myQueue.toString());
                    }catch(Exception e){
                        System.out.println(e.getMessage()+"\n");
                    } 
                }
            }
        }

        myQueue.enQueue(2);
        myQueue.enQueue(4);
        myQueue.enQueue(6);
        myQueue.enQueue(8);
        myQueue.enQueue(10);
        myQueue.enQueue(12);

        try{
            System.out.println("Empty: "+myQueue.isEmpty());
        }catch(Exception e){
            System.out.println(e.getMessage());
        } finally{
            try{
                System.out.println("Size: "+myQueue.size());
            }catch(Exception e){
                System.out.println(e.getMessage());
            } finally{
                try{
                    System.out.println("Dequeue: "+myQueue.deQueue());
                }catch(Exception e){
                    System.out.println(e.getMessage());
                } finally{
                    try{
                        System.out.println(myQueue.toString());
                    }catch(Exception e){
                        System.out.println(e.getMessage());
                    } 
                }
            }
        }

        while(!myQueue.isEmpty()){
            System.out.println("Dequeue: "+myQueue.deQueue());
        }

        try{
            System.out.println("Empty: "+myQueue.isEmpty());
        }catch(Exception e){
            System.out.println(e.getMessage());
        } finally{
            try{
                System.out.println("Size: "+myQueue.size());
            }catch(Exception e){
                System.out.println(e.getMessage());
            } finally{
                try{
                    System.out.println("Dequeue: "+myQueue.deQueue());
                }catch(Exception e){
                    System.out.println(e.getMessage());
                } finally{
                    try{
                        System.out.println(myQueue.toString());
                    }catch(Exception e){
                        System.out.println(e.getMessage());
                    } 
                }
            }
        }


    }

}